### ISSUE TITLE
This is excellent work, but I'm encountering an issue with the NeRF synthetic Lego dataset. The PLY file I obtained contains a lot of green dots. Could you please help me understand why this is happening and if there’s anything more I should do?

### ISSUE THREAD

**Issue by xzr52** [hbb1/2d-gaussian-splatting#146]
I ran the code with: 
                               python train.py -s data/nerf_synthetic/lego. 
During training at step 28600, I visualized the rendered image from the train.py script with the following code, resulting in an image with many green dots
            image, viewspace_point_tensor, visibility_filter, radii = render_pkg["render"], render_pkg["viewspace_points"], render_pkg["visibility_filter"], render_pkg["radii"]
            render_rgb_image = image.permute(1, 2, 0).detach().cpu().numpy()
            render_rgb_image = (render_rgb_image * 255).astype(np.uint8)
            render_rgb_image = Image.fromarray(render_rgb_image)
            render_rgb_image.save('render_rgb_image.png')
![image](https://github.com/user-attachments/assets/ec1c98a8-3f65-4e5c-9882-de0068434fc6)
I also ran ：
               python render.py -m output/af7f9997-f -s data/nerf_synthetic/lego 
to generate a PLY file, and visualizing this file also shows many green dots. 
![image](https://github.com/user-attachments/assets/a9a6dca3-92a4-450f-adee-0daf5f1d3873)
However, these green dots do not appear in the rendered image from render.py.
![image](https://github.com/user-attachments/assets/3b98f9c6-0194-4db5-88d7-e668758c98fc)
Could you help me understand why this is happening and if there's anything more I should do?

**Comment by hbb1** [comment]
I think I found the reason. We should clamp the output image to [0,1] before saving the images or for meshing.

**Comment by xzr52** [comment]
Thank you for your help; this method is indeed effective. However, I have another question: why does the exported mesh have a large area of black blocks at the back.Is it related to the mask?
![image](https://github.com/user-attachments/assets/5d3a5b8e-bfb6-4315-80a1-c58b533acb2b)

**Comment by hbb1** [comment]
I guess the reason is that the large gaussians are not effective culled, a minor issue introduced by [3DGS](https://github.com/graphdeco-inria/gaussian-splatting/issues/123).
